lib/common/procfs.c

```diff
@@ -140,3 +140,35 @@ crm_procfs_pid_of(const char *name)
     closedir(dp);
     return pid;
 }
+
+/*!
+ * \internal
+ * \brief Calculate number of logical CPU cores from procfs
+ *
+ * \param[out] cores  Where to store result
+ *
+ * \return pcmk_ok on success, -errno otherwise
+ */
+int
+crm_procfs_num_cores(int *cores)
+{
+    FILE *stream = NULL;
+
+    CRM_CHECK(cores != NULL, return -EINVAL);
+    *cores = 0;
+    stream = fopen("/proc/cpuinfo", "r");
+
+    if (stream == NULL) {
+        return -errno;
+    } else {
+        char buffer[256];
+
+        while (fgets(buffer, sizeof(buffer), stream)) {
+            if (strncmp(buffer, "processor", 9) == 0) {
+                ++*cores;
+            }
+        }
```

Comment:
Yes, gave it some more trial.  What we really need here is:
- read 9 chars from the beginning of the line and compare
- skip until the end of line if not present in those 9 chars already
- repeat
`fgets` does not handle the middle part for us, so it would have to be added by hand.